code/neuron_model.py

```diff
@@ -0,0 +1,307 @@
+
+    def I_Na(self, v, n):
+        """
+        Membrane current (in μA/cm²)
+        Sodium (Na)
+
+        |  :param v: Membrane voltage (mV)
+        |  :param n: Gating variable
+        |  :return: Sodium current
+        """
+        return (self.p['gNa'] * self.m_inf(v, self.p['sigma_Na'])**3 *
+                (0.85 - n) * (v - self.p['VNa']))
+
+    def I_NaP(self, v):
+        """
+        Membrane current (in μA/cm²)
+        Persistent Sodium (Na)
+
+        |  :param v: Membrane voltage (mV)
+        |  :param h: Gating variable
+        |  :return: Persistent sodium current
+        """
+        return (self.p['gNaP'] * self.m_inf(v, self.p['sigma_NaP'])**3 *
+                (v - self.p['VNa']))
+
+    def I_L(self, v):
+        """
+        Membrane current (in μA/cm²)
+        Leak current
+
+        |  :param v: Membrane voltage (mV)
+        |  :return: Leak current
+        """
+        return self.p['gL'] * (v - self.p['VL'])
+
+    def I_ext(self, period, duration, time):
+        """
+        External current (in μA/cm²)
+
+        |  :param period: Period of current on pulses
+        |  :param duration: Duration of current on pulses
+        |  :param amplitude: Pulse amplitude
+        |  :param time: Total simulation time (ms)
+        |  :param numChunks: Number of chunks according to period 
+        """
+        tmp = np.zeros((period, ))
+        tmp[:duration] = self.amplitude
+        # repeats = int(self.chunks + (time - period) / period)
+        repeats = int(self.chunks + (time / period))
+        return np.tile(tmp, repeats)
+
+    def rhs(self, t, X, Iapp):
+        """
+        Right-hand side of Wang [1] model.
+
+        |  :param X: States of the system
+        |  :param t: Time
+        |  :param Iapp: Injected current (stimulation)
+        |  :return: Calculate membrane potential & activation variables
+        """
+        V, h, H, n = X[0], X[1], X[2], X[3]
+
+        dV = ((-self.I_T(V, h) - self.I_H(V, H) - self.I_Na(V, n) -
+              self.I_K(V, n) - self.I_NaP(V) - self.I_L(V) +
+              Iapp)/self.p['Cm'])
+        dh = (self.p['phi_h']*(self.h_inf(V, self.p['theta_h'], self.p['k_h'])
+              - h)/self.tau_h(V, self.p['theta_h'], self.p['k_h']))
+        dH = self.p['phi_H'] * (self.H_inf(V) - H) / self.tau_H(V)
+        dn = (self.p['phi_K'] * (self.n_inf(V, self.p['sigma_K']) - n) /
+              self.tau_n(V, self.p['sigma_K']))
+
+        return np.array([dV, dh, dH, dn])
+
+    def run(self, tf, vname, tname, int_name='vode', *args):
+        """ 
+        Run the model by integrating equations.
+
+        |  :param tf: Total time (ms)
+        |  :param vname: Filename for storing membrane potential
+        |  :param tname: Filename for storing time
+        |  :param int_name: Integration method
+        |  :param *args: If int_method is vode, zvode, lsoda user can choose 
+                         Adams or BDF methods
```

Comment:
Documentation of `int_method` and `*args` is not general enough, mentions, e.g., not dopri5. Maybe the easiest solution is to reference the scipy.integrate.ode reference directly.